spec/unit/coinbase/user_spec.rb

```diff
@@ -74,38 +74,93 @@
+                                      'public_key': '0x1234567890',
+                                      'network_id': 'base-sepolia'
+                                    })
+    end
+
+    before do
+      allow(Coinbase::Client::AddressesApi).to receive(:new).and_return(addresses_api)
       allow(Coinbase::Client::WalletsApi).to receive(:new).and_return(wallets_api)
-      expect(wallets_api).to receive(:list_wallets).and_return(wallet_list)
-      expect(user.wallet_ids).to eq(wallet_ids)
+      expect(wallets_api)
+        .to receive(:list_wallets)
+        .and_return(
+          Coinbase::Client::WalletList.new({ 'data' => [wallet_model_1, wallet_model_2], 'total_count' => 2 })
+        )
+      expect(addresses_api)
+        .to receive(:list_addresses)
+        .with('wallet1', { limit: Coinbase::Wallet::MAX_ADDRESSES })
+        .and_return(
+          Coinbase::Client::AddressList.new({ 'data' => [address_model_1, address_model_2], 'total_count' => 2 })
+        )
+      expect(addresses_api)
+        .to receive(:list_addresses)
+        .with('wallet2', { limit: Coinbase::Wallet::MAX_ADDRESSES })
+        .and_return(
+          Coinbase::Client::AddressList.new({ 'data' => [address_model_3, address_model_4], 'total_count' => 2 })
+        )
+    end
+
+    it 'returns all wallets' do
+      wallets = user.wallets
+      expect(wallets.size).to eq(2)
+      expect(wallets[0].id).to eq(wallet_model_1.id)
+      expect(wallets[1].id).to eq(wallet_model_2.id)
     end
   end
 
   describe '#save_wallet_locally!' do
     let(:seed) { '86fc9fba421dcc6ad42747f14132c3cd975bd9fb1454df84ce5ea554f2542fbe' }
-    let(:address_count) { 1 }
-    let(:seed_wallet) do
-      Coinbase::Wallet.new(model, seed: seed, address_count: address_count)
-    end
-    let(:user) { described_class.new(model) }
-    Coinbase.configuration.backup_file_path = "#{SecureRandom.uuid}.json"
-    Coinbase.configuration.api_key_private_key = OpenSSL::PKey::EC.generate('prime256v1').to_pem
-    let(:wallet_id) { SecureRandom.uuid }
-    let(:addresses_api) { double('Coinbase::Client::AddressesApi') }
     let(:address_model) do
       Coinbase::Client::Address.new({
-                                      'address_id': '0xdeadbeef',
+                                      'address_id': '0xfbd9D61057eC1debCeEE12C62812Fb3E1d025201',
                                       'wallet_id': wallet_id,
                                       'public_key': '0x1234567890',
                                       'network_id': 'base-sepolia'
                                     })
     end
+    let(:wallet_id) { SecureRandom.uuid }
+    let(:seed_wallet) do
+      Coinbase::Wallet.new(model, seed: seed, address_models: [address_model])
+    end
+    let(:user) { described_class.new(model) }
+    Coinbase.configuration.backup_file_path = "#{SecureRandom.uuid}.json"
+    Coinbase.configuration.api_key_private_key = OpenSSL::PKey::EC.generate('prime256v1').to_pem
```

Comment:
This is a global test level mutation, if we're giong to do that can we move it to its own section?

Otherwise a common pattern is in a before block to set an instance variable to the existing value, update the value and in an `after` block return it to its original value. That avoids unnecessary test polution.

E.g. 
```ruby
    before do
      @backup_file_path = Coinbase.configuration.backup_file_path
      @api_key_private_key = Coinbase.configuration.api_key_private_key

      Coinbase.configuration.backup_file_path = "#{SecureRandom.uuid}.json"
      Coinbase.configuration.api_key_private_key = OpenSSL::PKey::EC.generate('prime256v1').to_pem

      allow(Coinbase::Client::AddressesApi).to receive(:new).and_return(addresses_api)
      File.open(Coinbase.configuration.backup_file_path, 'w') do |file|
        file.write(initial_seed_data)
      end
    end

    after do
      File.delete(Coinbase.configuration.backup_file_path)

      Coinbase.configuration.backup_file_path = @backup_file_path
      Coinbase.configuration.api_key_private_key = @api_key_private_key
    end
```

Reply:
Done.